File: Csharp/CodeWars/src/Resolutions/TheVowelCode.cs
namespace CodeWars.src.Resolutions;

public class TheVowelCode
{
    private static readonly string vowelForNumber = "aeiou";

    public static string Encode(string msg)
    {
        for (int i = 0; i < vowelForNumber.Length; i++)
            msg = msg.Replace(vowelForNumber[i].ToString(), (i + 1).ToString());

        return msg;
    }

    public static string Decode(string msg) =>
        string.Concat(msg.Select(x => char.IsDigit(x) ? vowelForNumber[(int)char.GetNumericValue(x) - 1] : x));
}

File: TypeScript/Codewars/src/resolutions/The-Vowel-Code.ts
const vowelForNumber: Record<number, string> = {
	1: "a",
	2: "e",
	3: "i",
	4: "o",
	5: "u",
};

export function encode(string: string): string {
	let encodedString = string;

	for (const [key, vowel] of Object.entries(vowelForNumber)) {
		encodedString = encodedString.replaceAll(vowel, key);
	}

	return encodedString;
}

export function decode(string: string): string {
	let encodedString = string;

	for (const [key, vowel] of Object.entries(vowelForNumber)) {
		encodedString = encodedString.replaceAll(key, vowel);
	}

	return encodedString;
}
